From 7d96c900d228456f3a2289de03b928ca9f5464ca Mon Sep 17 00:00:00 2001
Message-Id: <7d96c900d228456f3a2289de03b928ca9f5464ca.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Ken Lierman <ken.lierman@windriver.com>
Date: Thu, 21 Feb 2013 09:49:28 -0800
Subject: [PATCH 011/312] REVERTME [VPG]: drm/i915: Leave the brightness
 unchanged on enable if its 0

Android expects to control the state of the display via the brightness.
When a resume happens and the enable() function changes the brightness
to MAX, it turns the display on earlier than userspace expects, resulting
in odd visual behavior.

The actual initialization path uses the current state of the brightness
(set via firmware, etc), so the display will be initialized in a
good state without setting it to MAX here.

This was dropped from our tree at some point, so russell.webb@intel.com
reapplied this and amended the commit message only to mark as
MUST_REVERT and to modify the issue tag.  This is a hack that will be
reverted once VPG modifies user space to ensure the entire display stays
powered down when an alarm wakes the system from resume, but no screen
on wakelock is held.

Change-Id: I9b92e70c8e8fcc0b70c422ea44e6020aaff2812a
For: AXIA-4201
Orig-Change-Id: I7bdf741c510e3e9f3addab3db4246eec7704dabc
Signed-off-by: Russell Webb <russell.webb@intel.com>
(cherry picked from commit 5ef65c02c840b2a4f50f27ec04d7aa06ba206755)

Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/i915/intel_panel.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_panel.c b/drivers/gpu/drm/i915/intel_panel.c
index 38a9857..e217767 100644
--- a/drivers/gpu/drm/i915/intel_panel.c
+++ b/drivers/gpu/drm/i915/intel_panel.c
@@ -856,12 +856,14 @@ void intel_panel_enable_backlight(struct intel_connector *connector)
 
 	WARN_ON(panel->backlight.max == 0);
 
+#if 0
 	if (panel->backlight.level == 0) {
 		panel->backlight.level = panel->backlight.max;
 		if (panel->backlight.device)
 			panel->backlight.device->props.brightness =
 				panel->backlight.level;
 	}
+#endif
 
 	dev_priv->display.enable_backlight(connector);
 	panel->backlight.enabled = true;
-- 
1.7.9.5

